src/inapp/utils.ts

```diff
@@ -359,12 +359,11 @@ const mediaQueryXl = global?.matchMedia?.('(min-width: 1301px)');
 const generateSecuredIFrame = () => {
   const iframe = document.createElement('iframe');
   iframe.setAttribute('id', 'iterable-iframe');
-  // allow-popups and allow-top-navigation is to enable links for Safari since the iframe will block
-  // event handlers on elements in it preventing our custom link handling
-  iframe.setAttribute(
-    'sandbox',
-    'allow-same-origin allow-popups allow-top-navigation'
-  );
+  /**
+   * Most other browsers, if "allow-same-origin" enabled, already "allow-scripts". Safari does not by default, and same with some older browsers so
+   * we must manually set it here. See https://caniuse.com/mdn-html_elements_iframe_sandbox-allow-same-origin for more information.
+   */
+  iframe.setAttribute('sandbox', 'allow-same-origin allow-scripts');
```

Comment:
I think it's highly advised against to allow-scripts with allow-same-origin here, MDN docs for iframe says:

> When the embedded document has the same origin as the embedding page, it is strongly discouraged to use both allow-scripts and allow-same-origin, as that lets the embedded document remove the sandbox attribute — making it no more secure than not using the sandbox attribute at all.

We had to jump through a bunch of hoops for a security audit around JS execution so I could see adding this in being a problem. Unfortunately that was the reason we had to gate a bunch of functionality around Safari ☹️

Reply:
@ts-nguyen `allow-same-origin` allows for JS execution on newer browsers so if that's a security issue on Safari, then it's an issue with our non-safari implementation as well. If you look here https://caniuse.com/mdn-html_elements_iframe_sandbox-allow-same-origin , you'll see that other browsers allow script execution by default.